runtime/jcl/common/jclvm.c

```diff
@@ -325,3 +545,71 @@ Java_com_ibm_oti_vm_VM_dumpString(JNIEnv * env, jclass clazz, jstring str)
 		}
 	}
 }
+
+/**
+ * quicksort the list of pointers to statistics entries based on aggregateSize.
+ * @param list
+ * Vector of pointers to statistics entries
+ * @start index of the first element to be sorted
+ * @end index of the last element to be sorted
+ */
+
+static void
+sortStatsTable(J9HeapStatisticsTableEntry **list, IDATA start, IDATA end)
+{
+	IDATA leftScan = start;
+	IDATA rightScan = end;
+	IDATA pivot = (start + end) / 2;
+	UDATA pivotValue = list[pivot]->aggregateSize;
+	BOOLEAN leftMismatch = FALSE;
+	BOOLEAN rightMismatch = FALSE;
+	do {
+		while (!leftMismatch && (leftScan < rightScan)) {
+			if (list[leftScan]->aggregateSize <= pivotValue) {
+				leftMismatch = TRUE;
+			} else {
+				leftScan += 1;
+			}
+		}
+		while (!rightMismatch && (leftScan < rightScan)) {
+			if (list[rightScan]->aggregateSize >= pivotValue) {
+				rightMismatch = TRUE;
+			} else {
+				rightScan -= 1;
+			}
+		}
+		if (leftScan < rightScan) {
+			J9HeapStatisticsTableEntry *swap = list[rightScan];
+			list[rightScan] = list[leftScan];
+			list[leftScan] = swap;
+			leftMismatch = FALSE;
+			rightMismatch = FALSE;
+			leftScan += 1;
+			if (rightScan > leftScan) {
+				rightScan -= 1;
+			}
+		}
+	} while (leftScan < rightScan);
+
+	if ((end - start) >= 2) { /* list of length 1 or 2 is already sorted */
```

Comment:
This comment is still wrong: a list of length 2 might not be sorted.

Reply:
Then original list may not have been sorted, but it will be sorted after one pass through the loop above.  I will update the comment.